biogeochem/EDLoggingMortalityMod.F90

```diff
@@ -418,6 +433,237 @@ end subroutine get_harvest_rate_area
+     harvest_rate = 0._r8
+     harvest_rate_c = 0._r8
+     harvest_rate_supply = 0._r8
+     harvest_tag = 2
+
+     do h_index = 1,hlm_num_lu_harvest_cats
+        if (patch_anthro_disturbance_label .eq. primaryforest) then
+           if(hlm_harvest_catnames(h_index) .eq. "HARVEST_VH1") then! .or. &
+           !     hlm_harvest_catnames(h_index) .eq. "HARVEST_VH2") then
+              harvest_rate_c = harvest_rate_c + hlm_harvest_rates(h_index)
+              ! Determine the total supply of available C for harvest
+              if(harvestable_forest_c(h_index) >= harvest_rate_c) then
+                 harvest_rate_supply = harvest_rate_supply + harvestable_forest_c(h_index)
+                 harvest_tag(h_index) = 0
+              else
+                 harvest_tag(h_index) = 1
+              end if
+           endif
+        else if (patch_anthro_disturbance_label .eq. secondaryforest .and. &
+             secondary_age >= secondary_age_threshold) then
+           if(hlm_harvest_catnames(h_index) .eq. "HARVEST_SH1") then
+              harvest_rate_c = harvest_rate_c + hlm_harvest_rates(h_index)
+              if(harvestable_forest_c(h_index) >= harvest_rate_c) then
+                 harvest_rate_supply = harvest_rate_supply + harvestable_forest_c(h_index)
+                 harvest_tag(h_index) = 0
+              else
+                 harvest_tag(h_index) = 1
+              end if
+           endif
+        else if (patch_anthro_disturbance_label .eq. secondaryforest .and. &
+             secondary_age < secondary_age_threshold) then
+           if(hlm_harvest_catnames(h_index) .eq. "HARVEST_SH2") then! .or. &
+           !     hlm_harvest_catnames(h_index) .eq. "HARVEST_SH3") then
+              harvest_rate_c = harvest_rate_c + hlm_harvest_rates(h_index)
+              if(harvestable_forest_c(h_index) >= harvest_rate_c) then
+                 harvest_rate_supply = harvest_rate_supply + harvestable_forest_c(h_index)
+                 harvest_tag(h_index) = 0
+              else
+                 harvest_tag(h_index) = 1
+              end if
+           endif
+        endif
+     end do
+
+     ! If any harvest category available, assign to cur_harvest_tag and trigger logging event
+     if(present(cur_harvest_tag))then
+       cur_harvest_tag = minval(harvest_tag)
+       !write(fates_log(), *) 'cur_harvest_tag:', cur_harvest_tag
+       !write(fates_log(), *) 'harvest tags:', harvest_tag
+       !write(fates_log(), *) 'harvest rate c:', harvest_rate_c
+       !write(fates_log(), *) 'harvest rate supply:', harvest_rate_supply
+       !write(fates_log(), *) 'hlm harvest rates:', hlm_harvest_rates
+     end if
+
+     ! Transfer carbon-based harvest rate to area-based harvest rate
+     if (harvest_rate_supply > rsnbl_math_prec .and. harvest_rate_supply > harvest_rate_c) then
+        harvest_rate = harvest_rate_c / harvest_rate_supply
+     else
+        harvest_rate = 0._r8
+     end if
```

Comment:
so as I read the logic here, it is that if there is insufficient carbon to provide the harvest that the driving dataset is looking for on a given timestep, it will simply not harvest at all?  Would an alternative be to completely harvest the patch? I guess this is a minor effect but it seems like it might help simplify things by allowing the model to completely transfer all primary to secondary forest patches in a given place rather than keeping some tiny amount of secondary forest patch area around?

Reply:
Yes, it is a minor effect. I will modify this part.